library/core/src/slice/mod.rs

```diff
@@ -814,7 +814,7 @@ impl<T> [T] {
     #[stable(feature = "rust1", since = "1.0.0")]
     #[inline]
     pub fn chunks(&self, chunk_size: usize) -> Chunks<'_, T> {
-        assert_ne!(chunk_size, 0);
+        assert_ne!(chunk_size, 0, "Chunks cannot have a size of zero!");
```

Comment:
```suggestion
        assert_ne!(chunk_size, 0, "chunks cannot have a size of zero");
```
There is no reason to yell.

Reply:
I felt and feel like a failed assertion has plenty of reason to yell, but either way is a big improvement over no message, which is what I really care about.

Reply:
The assertion is after all some kind of error message. And why would yell at someone in the error messages. There is no reason for it as we can gently explain what's wrong.

Reply:
The same reason why you would shout "Watch out for that car!" instead of saying it gently; because it is more likely to be noticed. I feel that an assertion failure is more serious than just an error message, because it is unexpected and so you may not be paying much attention.

But I am sure there are conventions for this sort of thing already and either way would make me happy. The important part is that there should be some message instead of no message.